Userland/Libraries/LibIPC/Connection.cpp

```diff
@@ -49,11 +49,20 @@ ErrorOr<void> ConnectionBase::post_message(MessageBuffer buffer)
 #endif
 
     ReadonlyBytes bytes_to_write { buffer.data.span() };
+    int writes_done = 0;
+    size_t initial_size = bytes_to_write.size();
     while (!bytes_to_write.is_empty()) {
         auto maybe_nwritten = m_socket->write(bytes_to_write);
+        writes_done++;
         if (maybe_nwritten.is_error()) {
             auto error = maybe_nwritten.release_error();
             if (error.is_errno()) {
+                // FIXME: This is a hacky way to at least not crash on large messages
+                // The limit of 100 writes is arbitrary, and there to prevent indefinite spinning on the EventLoop
+                if (error.code() == EAGAIN && writes_done < 100) {
+                    sched_yield();
```

Comment:
Don't you want to yield to the event loop? that'd be `Core::EventLoop::current().pump()`.

Reply:
Yes, but actually no. That's the way it should eventually be, but that would probably need extra logic I didn't want to work on now.

Reply:
Yes, but actually no.
Something along those lines would be better, but this is basically just a hotfix that works well enough for now.